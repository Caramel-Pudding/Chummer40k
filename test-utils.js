import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore, combineReducers } from "@reduxjs/toolkit";
import { Provider } from "react-redux";

import characteristics from "./redux/features/characteristics/slice";
import descriptors from "./redux/features/descriptors/slice";

// Import your own reducer
const reducer = combineReducers({ descriptors, characteristics });

function render(
  ui,
  {
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    initialState,
    store = createStore(reducer, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";
// override render method
export { render };
