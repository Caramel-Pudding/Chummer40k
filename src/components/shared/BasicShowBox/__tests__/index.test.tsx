import { render, cleanup } from "@/tests/test-utils";

import { BasicShowBox } from "..";

describe("<BasicShowBox />", () => {
  afterEach(cleanup);

  it("Basic snapshot test", () => {
    // * #TEST: ARRANGE
    const { container } = render(<BasicShowBox text="TEST" />, {});
    // * #TEST: ASSERT
    expect(container).toMatchSnapshot();
  });
});
