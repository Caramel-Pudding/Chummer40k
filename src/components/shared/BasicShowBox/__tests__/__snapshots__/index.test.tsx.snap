// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicShowBox /> Basic snapshot test 1`] = `
<div>
  <button
    class="p-1 border border-black cursor-pointer"
    type="button"
  >
    TEST
  </button>
</div>
`;
